spec/hbw/features/tasks/search_spec.rb

```diff
@@ -0,0 +1,70 @@
+feature 'Check searching in table with tasks', js: true do
+  before(:each) do
+    user = FactoryBot.create(:user)
+
+    signin(user.email, user.password)
+    expect(page).not_to have_content 'Sign in'
+    expect(page).to     have_content 'Orders list'
+
+    ajax { click_on 'Tasks' }
+    expect_widget_presence
+    expect(page).to have_content 'Open tasks'
+  end
+
+  describe 'search field rendered' do
+    it 'with too short search query' do
+      fill_search_field('te')
+      expect_widget_presence
+    end
+
+    it 'with valid search request' do
+      expect(tasks_table_content).to eq(
+        [
+          ['Medium', 'Check test form', ' Test name', 'No description', 'expired'],
+          ['High', 'Check test form', ' Test name', 'Some test description', '30 Jun 2016']
+        ]
+      )
+
+      fill_search_field('description')
+
+      expect(search_field_text).to eq('description')
+      expect(tasks_table_content).to eq(
+        [
+          ['High', 'Check test form', ' Test name', 'Some test description', '30 Jun 2016']
+        ]
+      )
+    end
+
+    it 'with clearing after tab changes' do
+      fill_search_field('description')
+      expect(search_field_text).to eq('description')
+
+      click_on_tab 'Unassigned tasks (10)'
+
+      expect(search_field_text).to eq('')
+    end
+
+    it 'with clearing search query on cross click' do
+      expect(tasks_table_content).to eq(
+        [
+          ['Medium', 'Check test form', ' Test name', 'No description', 'expired'],
+          ['High', 'Check test form', ' Test name', 'Some test description', '30 Jun 2016']
+        ]
+      )
+
+      fill_search_field('description')
+
+      expect(search_field_text).to eq('description')
+
+      clear_search_field
+
+      expect(search_field_text).to eq('')
+      expect(tasks_table_content).to eq(
+        [
+          ['Medium', 'Check test form', ' Test name', 'No description', 'expired'],
+          ['High', 'Check test form', ' Test name', 'Some test description', '30 Jun 2016']
```

Comment:
well, you just checked that nothing has been changed :)

Reply:
Thats the point. After clearing search field everything has to be as before search.

Reply:
But you should check that clear cross will change state when clicked.